xbmc/music/tags/MusicInfoTag.cpp

```diff
@@ -433,11 +433,11 @@ void CMusicInfoTag::SetYear(int year)
   // Parse integer year value into YYYY ISO8601 format (partial) date string
   // Add century for to 2 digit numbers, 41 -> 1941, 40 -> 2040
   if (year > 99)
-    SetReleaseDate(StringUtils::Format("{:04}", year));
+    SetReleaseDate(fmt::format("{0:4}", year));
```

Comment:
This seems wrong

Reply:
According to the API docs for fmt, it follows the same structure as Python.  If I leave the code as `{:04}` then I get a year of 1865 when no year value is present.  If I change it to `{0:4}` (which is how Python does it) I get an empty year, which is correct.  Ofc I could be wrong, but I have tested both ways several times although only with the one set of files.

Reply:
Maybe it's not really clear what the desired behavior is here but the leading `0` before the colon `:` just refers to the id of the parameter. The values after the colon `:` refer to the format spec, in this case it means 4 characters width with `0` values padded at the start (to enforce 4 character width).

If you want no year when no year is present I would suggest adding logic to fix that.

Reply:
> If you want no year when no year is present I would suggest adding logic to fix that.

Yes the logic already does that. What is wanted is a 4 digit value - YYYY given any value > 0, with centuary wrapping for YY values. Mainly it is handling just 2 digit and 4 digit year values, but  zero padding any odd 3 digit value is fine. `{:04}` does that.

However looking at it there is a flaw in the existing code for single digit years. If input 1 then release date is set to "0201", what is wanted is "2001"

Reply:
My mistake.  Unexpected dates were from scraped data, not scanned. Sorry for the noise. Done.